src/main/java/org/springframework/data/elasticsearch/core/RequestFactory.java

```diff
@@ -1014,7 +1019,9 @@ private SearchRequest prepareSearchRequest(Query query, @Nullable Class<?> clazz
 			sourceBuilder.searchAfter(query.getSearchAfter().toArray());
 		}
 
-		query.getRescorerQueries().forEach(rescorer -> sourceBuilder.addRescorer(getQueryRescorerBuilder(rescorer)));
+		for (RescorerQuery rescorer : query.getRescorerQueries()) {
+			sourceBuilder.addRescorer(getQueryRescorerBuilder(rescorer));
+		}
```

Comment:
please do not refactor code unrelated to the issue that is solved

Reply:
this needed to be done because of line 944. If a lambda is used then `sourceBuilder` needs to be `final`. I think I can write a method to accomplish this though, I will make the change